src/test_reference.jl

```diff
@@ -122,7 +122,7 @@ function test_reference(
     end
 
     # file exists
-    reference = loadfile(T, file)
+    reference = loadfile(typeof(actual), file)
```

Comment:
I suspect we always should have been doing this  (for SHA) and that we were just missing the tests that showed it was needed.